analyzers/src/SonarAnalyzer.CSharp/Rules/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -18,144 +18,106 @@
-        }
+                var objectCreation = (ObjectCreationExpressionSyntax)c.Node;
+                CheckCall(c, objectCreation, objectCreation.ArgumentList);
+            },
+            SyntaxKind.ObjectCreationExpression);
+    }
 
-        private static bool IsInvocationWithExplicitArray(ArgumentListSyntax argumentList, IMethodSymbol invokedMethodSymbol, SemanticModel semanticModel)
+    private static void CheckCall(SonarSyntaxNodeReportingContext context, SyntaxNode node, ArgumentListSyntax argumentList)
+    {
+        if (argumentList is { Arguments.Count: > 0 }
+            && context.SemanticModel.GetSymbolInfo(node).Symbol is IMethodSymbol method
+            && method.Parameters.LastOrDefault() is { IsParams: true }
+            && !IsInvocationWithExplicitArray(argumentList, method, context.SemanticModel)
+            && ArgumentTypes(context, argumentList) is var argumentTypes
+            && argumentTypes.All(x => x is not IErrorTypeSymbol)
+            && OtherOverloadsOf(method).FirstOrDefault(x =>
+                ArgumentsMatchParameters(argumentList, argumentTypes, x, context.SemanticModel) && MethodAccessibleWithinType(x, context.ContainingSymbol.ContainingType))
+                    is { } otherMethod)
         {
-            var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, invokedMethodSymbol);
-
-            var allParameterMatches = new List<IParameterSymbol>();
-            foreach (var argument in argumentList.Arguments)
-            {
-                if (!methodParameterLookup.TryGetSymbol(argument, out var parameter))
-                {
-                    return false;
-                }
-
-                allParameterMatches.Add(parameter);
-
-                if (!parameter.IsParams)
-                {
-                    continue;
-                }
-
-                var argType = semanticModel.GetTypeInfo(argument.Expression).Type;
-                if (argType is not IArrayTypeSymbol)
-                {
-                    return false;
-                }
-            }
-
-            return allParameterMatches.Count(p => p.IsParams) == 1;
+            context.ReportIssue(Diagnostic.Create(Rule, node.GetLocation(), otherMethod.ToMinimalDisplayString(context.SemanticModel, node.SpanStart)));
         }
+    }
 
-        private static bool ArgumentsMatchParameters(ArgumentListSyntax argumentList, List<INamedTypeSymbol> argumentTypes, IMethodSymbol possibleOtherMethod, SemanticModel semanticModel)
-        {
-            var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, possibleOtherMethod);
+    private static List<ITypeSymbol> ArgumentTypes(SonarSyntaxNodeReportingContext context, ArgumentListSyntax argumentList) =>
+        argumentList.Arguments
+            .Select(x => context.SemanticModel.GetTypeInfo(x.Expression))
+            .Select(x => x.Type ?? x.ConvertedType) // Action and Func won't always resolve properly with Type
+            .ToList();
```

Comment:
We will never need to add more items to it.
```suggestion
            .ToArray();
```